### PR TITLE
Autosubscribe behavior

### PR DESCRIPTION
**Pull request by timbussmann**
Documents the new autosubscribe behavior and the `DisableFor` API.

### CODE REVIEW

**Comment by mauroservienti** [comment]
Should we mention somewhere the type of error message that'll be logged?

File: nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed_exclude-event-types_[6,).partial.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 ### Exclude event types from auto subscribe
 
+If a message handler is defined for an event but no publishers information can be found the endpoint will log an error, as the following one, at startup:
```
Comment (timbussmann) [inline nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed_exclude-event-types_[6,).partial.md]: I'm not sure we need to document the log entry? Also that warning doesn't strictly belong into the `Exclude event types from auto subscribe` section imo.

**Comment by mauroservienti** [comment]
@dbelcham @bording can you please review these changes as Core maintainers?

File: nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed.md [hunk 2]
```diff
@@ -16,7 +16,17 @@ Messages matching the following criteria will be auto subscribed at startup.
 
  1. Defined as an event either using `IEvent` or by the `.DefiningEventsAs` convention.
  1. At least one [message handler and/or saga](/nservicebus/handlers/) exists for the given event.
- 1. If the selected transport [does not support publish-subscribe natively](/transports/#types-of-transports-unicast-only-transports), the publisher for that message needs to be specified via the [routing](/nservicebus/messaging/routing.md) API.
+
+If a message handler is defined for an event but no publishers information can be found the endpoint will log an error, as the following one, at startup:
```
Comment (bording) [inline nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed.md]: I think this section needs to be changed a bit. The only time this warning is relevant is for message-based pubsub, so this mention of the warning should come after the discussion about needing to use `RegisterPublisher`. Then, you can point out that if `RegisterPublisher` hasn't been called, the warning will be logged.

File: Snippets/Core/Core_6/Core_6.csproj [hunk 3]
```diff
@@ -7,7 +7,7 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.ServiceFabric.Services" Version="2.*" />
-    <PackageReference Include="NServiceBus" Version="6.*" />
+    <PackageReference Include="NServiceBus" Version="6.5-*" />
```
Comment (bording) [inline Snippets/Core/Core_6/Core_6.csproj]: This is going to mean there will need to be a prerelease.txt in the snippets folder, but then that will also mean we'll end displaying a prerelease warning on all v6 pages.

I think we might need to split this out in some other way. @DavidBoike Any ideas?

File: Snippets/Core/Core_7/Core_7.csproj [hunk 4]
```diff
@@ -7,7 +7,7 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.ServiceFabric.Services" Version="2.*" />
-    <PackageReference Include="NServiceBus" Version="7.*" />
+    <PackageReference Include="NServiceBus" Version="7.1-*" />
```
Comment (bording) [inline Snippets/Core/Core_7/Core_7.csproj]: Same prerelease.txt problem here.

File: Snippets/Core/Core_6/Core_6.csproj (hunk 3 above)
Comment (DavidBoike) [reply Snippets/Core/Core_6/Core_6.csproj]: I think there has to be a Core_6.5 folder temporarily.

File: nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed.md (hunk 2 above)
Comment (mauroservienti) [reply nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed.md]: Yeah, better.

File: Snippets/Core/Core_6/Core_6.csproj (hunk 3 above)
Comment (mauroservienti) [reply Snippets/Core/Core_6/Core_6.csproj]: I’ll give it a try next week.

**Comment by mauroservienti** [comment]
Merging as is. I’ll have a look at the rendered result and amend if needed.

**Pull request merged**